src/replication_handler/replication_handler.cpp

```diff
@@ -268,8 +269,27 @@ auto ReplicationHandler::GetRole() const -> memgraph::replication_coordination_g
   return repl_state_.GetRole();
 }
 
+auto ReplicationHandler::GetDatabasesHistories() -> replication_coordination_glue::DatabaseHistories {
+  replication_coordination_glue::DatabaseHistories results;
+  dbms_handler_.ForEach([&results](memgraph::dbms::DatabaseAccess db_acc) {
+    auto *storage = db_acc->storage();
+    auto &repl_storage_state = storage->repl_storage_state_;
+
+    std::vector<std::pair<std::string, uint64_t>> history =
```

Comment:
```suggestion
    auto history =
```
But I don't see any additional processing you just return pair as it is

Reply:
why even fmap it? just use return value from history

Reply:
Value in history is deque